### PR TITLE
Fix a bug that fromData doesn't add supplied data

### PR DESCRIPTION
**Pull request by touhonoob**

### CODE REVIEW

**Comment by kevmoo** [comment]
Please rebase on master – I just did some tweaks, too!

**Comment by touhonoob** [comment]
@kevmoo rebased

File: example/src/qr_demo.dart [hunk 1]
```diff
@@ -70,7 +70,7 @@ class QrDemo {
     //
     // Type Div
     //
-    for (var i = 1; i <= 10; i++) {
+    for (var i = 1; i <= 40; i++) {
```
Comment (kevmoo) [inline example/src/qr_demo.dart]: keep this at 10 – 40 is WAAAY too big. breaks the UI

File: test/qr_code_test_data.dart [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import 'dart:convert';
 
-var qrCodeTestData = jsonDecode(r'''
+final dynamic qrCodeTestData = jsonDecode(r'''
```
Comment (kevmoo) [inline test/qr_code_test_data.dart]: no need for `dynamic`!

Comment (touhonoob) [reply test/qr_code_test_data.dart]: removed

File: example/src/qr_demo.dart (hunk 1 above)
Comment (touhonoob) [reply example/src/qr_demo.dart]: reverted

File: lib/src/qr_code.dart [hunk 3]
```diff
@@ -19,6 +19,8 @@ class QrCode {
   List<int> _dataCache;
   final List<QrByte> _dataList = <QrByte>[];
 
+  List<List<bool>> get modules => _modules;
```
Comment (kevmoo) [inline lib/src/qr_code.dart]: Does this *need* to be public for any use case other than testing?

If yes, just make the final field public...

Comment (touhonoob) [reply lib/src/qr_code.dart]: added `@visibleForTesting` annotation

**Pull request merged**